src/table.js

```diff
@@ -666,3 +667,90 @@ export function __table(source, operations) {
   }
   return source;
 }
+
+function isValidSchema(schema) {
+  if (!schema || !Array.isArray(schema)) return;
+  return schema.every((s) => {
+    s && typeof s.name === "string" && typeof s.type === "string";
+  });
+}
+
+function initKey() {
+  return {
+    other: 0,
+    boolean: 0,
+    integer: 0,
+    number: 0,
+    date: 0,
+    string: 0,
+    array: 0,
+    object: 0,
+    bigint: 0, // TODO for csv, tsv?
+    buffer: 0
+  };
+}
+
+function inferSchema(source) {
+  const schema = [];
+  const sampleSize = 100;
+  let sample = source.slice(0, sampleSize);
+  if (arrayIsPrimitive(sample)) {
+    sample = sample.map((d) => {
+      return {value: d};
+    });
+  }
+  const typeCounts = {};
+  sample.map((d) => {
```

Comment:
The return value of sample.map is being ignored here. We should prefer a for-of loop instead as the standard mechanism for iteration (and if not that, then sample.forEach, but I usually try to use for-of for consistency).